### DISCUSSION TITLE
Add local CA certificates when Docker image runs

### DISCUSSION THREAD

**Discussion by Xoffio** [supabase/supabase#20607] [Feature Requests]
Hi,

While working with Supabase, I ran into something that might happen to some users hosting Supabase locally.

In my setup, I needed to integrate with a local SMTP server that redirects emails to Gmail. To make this work within my internal network, I need to add a CA certificate to the services talking to my SMTP server.

In the case of hosting Supabase locally, I was able to do this by manually editing the Dockerfile of the `gotrue` image:

```Dockerfile
COPY ./certs/* /usr/local/share/ca-certificates/
RUN update-ca-certificates
```

and then editing the `docker-compose.yml` file of supabase:

```yml
  auth:
    container_name: supabase-auth
    image: gotrue #supabase/gotrue:v2.132.3
```

## Problem solution suggestion
I suggest introducing a `certs` volume in Supabase where users can conveniently place their certificates. This can be accompanied by a script that runs during the Docker image startup, copying the certificates to `/usr/local/share/ca-certificates/` and updating the CA certificate seamlessly.

This would simplify the process of incorporating custom certificates into the Supabase Docker image, making it more user-friendly for diverse setups.

**Comment by Stokestack** [comment]
I'm losing a huge amount of time looking for information on how to get self-generated certs working for local development (in my case the CLI local development, not a full local installation I guess).

There doesn't seem to be any documentation, and so far no answers on Discord or Supabase issues. The only suggestions so far online were to route traffic through a third-party service online, like Cloudflare Tunnels.

This is such a fundamental and important problem to resolve. I'm developing a mobile app, and those must use HTTPS, per app-store rules.
